crates/nu-cli/src/completions/custom_completions.rs

```diff
@@ -1,19 +1,127 @@
+                    start: span.start - offset,
+                    end: span.end - offset,
+                },
+                ..Suggestion::default()
+            };
+            let mut value_type = Type::String;
+
+            // Iterate the cols looking for `value` and `description`
+            record.iter().for_each(|(key, value)| {
+                match key.as_str() {
+                    "value" => {
+                        value_type = value.get_type();
+                        // Convert the value to string
+                        if let Ok(val_str) = value.coerce_string() {
+                            // Update the suggestion value
+                            suggestion.value = val_str;
+                        }
+                    }
+                    "description" => {
+                        // Convert the value to string
+                        if let Ok(desc_str) = value.coerce_string() {
+                            // Update the suggestion value
+                            suggestion.description = Some(desc_str);
+                        }
+                    }
+                    "style" => {
+                        // Convert the value to string
+                        suggestion.style = match value {
+                            Value::String { val, .. } => Some(lookup_ansi_color_style(val)),
+                            Value::Record { .. } => Some(color_record_to_nustyle(value)),
+                            _ => None,
+                        };
+                    }
+                    "span" => {
+                        if let Value::Record { val: span, .. } = value {
+                            let start = span
+                                .get("start")
+                                .and_then(|val| val.as_int().ok())
+                                .and_then(|x| usize::try_from(x).ok());
+                            let end = span
+                                .get("end")
+                                .and_then(|val| val.as_int().ok())
+                                .and_then(|x| usize::try_from(x).ok());
+                            // TODO: error on invalid spans?
+                            match (start, end) {
+                                (Some(start), Some(end)) => {
+                                    suggestion.span = reedline::Span {
+                                        start: start.min(end),
+                                        end,
+                                    };
+                                }
+                                (Some(start), None) => {
+                                    suggestion.span.start = start.min(suggestion.span.end);
+                                }
+                                (None, Some(end)) => {
+                                    suggestion.span.end = end.max(suggestion.span.start);
+                                }
+                                _ => (),
+                            }
```

Comment:
I moved `map_value_completions` to this file since it's the only place calling that function.

And this is the span related change.